script.js

```diff
@@ -0,0 +1,333 @@
+      return -this.firstElementChild.offsetLeft + this.margin;
+    }
+    return newTranslate;
+  };
+
+  /**
+   * Handle focus from spacial navigator
+   */
+  proto._handleFocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._scrollTo(itemElem);
+    this._fireEvent('focus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+
+    // Since we may have many elements in one node, active-node-changed event
+    // happens when active node is changed.
+    if (this.currentNode && this.currentNode !== nodeElem) {
+      this._fireEvent('active-node-changed', {
+        scrollbox: this,
+        oldNodeElem: this.currentNode,
+        newNodeElem: nodeElem
+      });
+    }
+    this.currentNode = nodeElem;
+    itemElem.focus();
+  };
+
+  /**
+   * Handle unfocus from spacial navigator
+   */
+  proto._handleUnfocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._fireEvent('blur', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+    itemElem.blur();
+  };
+
+  /**
+   * Add/Remove elements to/from spacial navigator when a child is
+   * added/removed.
+   */
+  proto._handleMutation = function(mutations) {
+    mutations.forEach(function(mutation) {
+      var prevElem = mutation.previousSibling;
+      var nextElem = mutation.nextSibling;
+      if (prevElem) {
+        prevElem = (prevElem.nodeType === Node.ELEMENT_NODE)?
+                                    prevElem : prevElem.previousElementSibling;
+      }
+      if (nextElem) {
+        nextElem = (nextElem.nodeType === Node.ELEMENT_NODE)?
+                                    nextElem : nextElem.nextElementSibling;
+      }
+
+      var nodes;
+      if (mutation.type === 'childList') {
+        nodes = Array.prototype.slice.call(mutation.addedNodes);
+        // add every element to spacial navigator
+        nodes.forEach(this._onNodeAdded, this);
+
+        nodes = Array.prototype.slice.call(mutation.removedNodes);
+        // remove every element from spacial navigator
+        nodes.forEach(function(node) {
+          this._onNodeRemoved(node, prevElem, nextElem);
+        }, this);
+      }
+    }, this);
+  };
+
+  proto._onNodeAdded = function(node) {
+    var navigableElems = Array.prototype.slice.call(
+                                  node.getElementsByClassName('navigable'));
+    if (node.classList.contains('navigable')) {
+      navigableElems.push(node);
+    }
+    this._spatialNavigator.multiAdd(navigableElems);
+    this._fireEvent('node-added', {
+      node: node
+    });
+  };
+
+  proto._onNodeRemoved = function(node, prevElem, nextElem) {
+    var navigableElems = Array.prototype.slice.call(
+                                  node.getElementsByClassName('navigable'));
+    if (node.classList.contains('navigable')) {
```

Comment:
Is there any requirement that we need to allow multiple `navigable` inside one node?